{"uid":"85c5838b34256cc6","name":"test_simulate_s3_service_failure","fullName":"tests.test_aws_pipeline#test_simulate_s3_service_failure","historyId":"90d8c180a6d43e94d9ce09d0d41f52d5","time":{"start":1753178144632,"stop":1753178157125,"duration":12493},"status":"failed","statusMessage":"AssertionError: 4. Failure - S3 FILE UPLOAD FAILED: File not found\nassert 'File not found' in 'Successfully uploaded filtered data to s3://lambda-filtered-output-bucket/filtered_data/2025-07-22T09-55-45-956719Z.json\\n'","statusTrace":"def test_simulate_s3_service_failure():\n        config = get_config()\n    \n        # take the returned response from fetch guardian article function\n        guardian_api_response = get_guardian_articles(config['CATEGORY'])\n        assert guardian_api_response.status_code == 200, (\n            f'1. Failure - GET API FAILED: {guardian_api_response.status_code}'\n        )\n        print('1. Success - GET API PASSED')\n    \n        # Send the data to the SQS queue\n        message = guardian_api_response.json()\n        sqs_response = send_message_to_sqs(message)\n        sqs_response_status = sqs_response['ResponseMetadata']['HTTPStatusCode']\n        assert sqs_response_status == 200, (\n            f'2. Failure - SQS QUEUE FAILED: {sqs_response_status}'\n        )\n        print('2. Success - SQS QUEUE PASSED')\n    \n        # verify lambda is triggered from sqs\n        time.sleep(10)\n        lambda_status, sqs_message, s3_message = check_lambda_logs_in_cloud_watch(fail_on_purpose=False)\n        assert lambda_status == True, (f'3. Failure - LAMBDA FUNCTION FAILED:\\nSQS:{sqs_message}\\nS3:{s3_message}')\n        print('3. Success - LAMBDA FUNCTION PASSED')\n    \n        # verify if the filtered data is uploaded to S3 or not\n        latest_s3_uploaded_file = get_latest_file_from_s3(fail_on_purpose=True)\n>       assert latest_s3_uploaded_file in s3_message, (f'4. Failure - S3 FILE UPLOAD FAILED: {latest_s3_uploaded_file}')\nE       AssertionError: 4. Failure - S3 FILE UPLOAD FAILED: File not found\nE       assert 'File not found' in 'Successfully uploaded filtered data to s3://lambda-filtered-output-bucket/filtered_data/2025-07-22T09-55-45-956719Z.json\\n'\n\ntests/test_aws_pipeline.py:96: AssertionError","flaky":false,"newFailed":false,"newBroken":false,"newPassed":false,"retriesCount":0,"retriesStatusChange":false,"beforeStages":[],"testStage":{"status":"failed","statusMessage":"AssertionError: 4. Failure - S3 FILE UPLOAD FAILED: File not found\nassert 'File not found' in 'Successfully uploaded filtered data to s3://lambda-filtered-output-bucket/filtered_data/2025-07-22T09-55-45-956719Z.json\\n'","statusTrace":"def test_simulate_s3_service_failure():\n        config = get_config()\n    \n        # take the returned response from fetch guardian article function\n        guardian_api_response = get_guardian_articles(config['CATEGORY'])\n        assert guardian_api_response.status_code == 200, (\n            f'1. Failure - GET API FAILED: {guardian_api_response.status_code}'\n        )\n        print('1. Success - GET API PASSED')\n    \n        # Send the data to the SQS queue\n        message = guardian_api_response.json()\n        sqs_response = send_message_to_sqs(message)\n        sqs_response_status = sqs_response['ResponseMetadata']['HTTPStatusCode']\n        assert sqs_response_status == 200, (\n            f'2. Failure - SQS QUEUE FAILED: {sqs_response_status}'\n        )\n        print('2. Success - SQS QUEUE PASSED')\n    \n        # verify lambda is triggered from sqs\n        time.sleep(10)\n        lambda_status, sqs_message, s3_message = check_lambda_logs_in_cloud_watch(fail_on_purpose=False)\n        assert lambda_status == True, (f'3. Failure - LAMBDA FUNCTION FAILED:\\nSQS:{sqs_message}\\nS3:{s3_message}')\n        print('3. Success - LAMBDA FUNCTION PASSED')\n    \n        # verify if the filtered data is uploaded to S3 or not\n        latest_s3_uploaded_file = get_latest_file_from_s3(fail_on_purpose=True)\n>       assert latest_s3_uploaded_file in s3_message, (f'4. Failure - S3 FILE UPLOAD FAILED: {latest_s3_uploaded_file}')\nE       AssertionError: 4. Failure - S3 FILE UPLOAD FAILED: File not found\nE       assert 'File not found' in 'Successfully uploaded filtered data to s3://lambda-filtered-output-bucket/filtered_data/2025-07-22T09-55-45-956719Z.json\\n'\n\ntests/test_aws_pipeline.py:96: AssertionError","steps":[],"attachments":[{"uid":"c6806ea7624acd3f","name":"stdout","source":"c6806ea7624acd3f.txt","type":"text/plain","size":307}],"parameters":[],"shouldDisplayMessage":true,"stepsCount":0,"attachmentsCount":1,"hasContent":true,"attachmentStep":false},"afterStages":[],"labels":[{"name":"parentSuite","value":"tests"},{"name":"suite","value":"test_aws_pipeline"},{"name":"host","value":"pkrvmq0rgcvqdmg"},{"name":"thread","value":"2463-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"tests.test_aws_pipeline"},{"name":"resultFormat","value":"allure2"}],"parameters":[],"links":[],"hidden":false,"retry":false,"extra":{"severity":"normal","retries":[],"categories":[{"name":"Product defects","matchedStatuses":[],"flaky":false}],"tags":[]},"source":"85c5838b34256cc6.json","parameterValues":[]}